gradle/plugins/com.liferay.ide.gradle.core/src/com/liferay/ide/gradle/core/GradleUtil.java

```diff
@@ -55,4 +70,80 @@ public static Job importGradleProject( File dir , IProgressMonitor monitor) thro
 
         return synchronizeGradleProjectJob;
     }
+
+    public static Job executeTask(final IProject project , final String task)
```

Comment:
Lets make this method synchroniuous and not create a job but directly execute the task.  The calling method can add a job if it needs to.